### PR TITLE
Remove redhat domain from domain exclusion

### PR DESCRIPTION
**Pull request by rajivnathan**
redhat accounts should not be exempt from deactivation on production.

This should only be merged after https://github.com/codeready-toolchain/host-operator/pull/345

Related issue:
https://issues.redhat.com/browse/CRT-874

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]
Looks good but we have to make sure we set `deactivation-domains-excluded: '@redhat.com'` in our crt-stage because we do want to keep redhat accounts active in crt-stage.

**Comment by alexeykazakov** [comment]
> Looks good but we have to make sure we set deactivation-domains-excluded: '@redhat.com' in our crt-stage because we do want to keep redhat accounts active in crt-stage.

Done ^^^

File: deploy/env/prod.yaml [hunk 1]
```diff
@@ -22,4 +22,4 @@ host-operator:
     name: host-operator-secret
   config-map:
     name: host-operator-config
-  deactivation-domains-excluded: '@redhat.com'
+  deactivation-domains-excluded: ''
```
Comment (alexeykazakov) [inline deploy/env/prod.yaml:25]:
```suggestion
```

**Review by MatousJobanek — APPROVED** [review]

**Comment by MatousJobanek** [comment]
@rajivnathan shouldn't we change the `deactivationTimeoutDays` at the basic tier to 30 days before we merge this commit? Otherwise, it will deactivate all users that are already provisioned to prod.
https://github.com/codeready-toolchain/host-operator/blob/master/deploy/templates/nstemplatetiers/basic/tier.yaml#L14

**Comment by rajivnathan** [comment]
@MatousJobanek good point. Done in https://github.com/codeready-toolchain/host-operator/pull/338/commits/d668eb6d5c690eebc6a6ae89e810580304cb3a61

**Comment by rajivnathan** [comment]
I see that the basic tier in prod now has `deactivationTimeoutDays: 30` so we are good to merge this PR

**Pull request merged**